### ISSUE TITLE
Module not found: Can't resolve 'apollo-server-errors'

### ISSUE THREAD

**Issue by snikch** [confuser/graphql-constraint-directive#156]
When importing this package into a next.js app, I get the following error. I believe because https://github.com/confuser/graphql-constraint-directive/blob/master/index.js#L135 attempts to `require('apollo-server-errors')` which is not a dependency of this package.

```
../../.yarn/__virtual__/graphql-constraint-directive-virtual-74b5cc573a/0/cache/graphql-constraint-directive-npm-4.1.2-695ac942e4-d61673215b.zip/node_modules/graphql-constraint-directive/index.js:135:39
Module not found: Can't resolve 'apollo-server-errors'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
../../packages/graphql/schemas/scalars/index.ts
../../packages/graphql/index.ts
./pages/api/graphql.ts
```

I'm not really sure what an appropriate resolution here is, short of moving the individual library support to alternative packages.

**Comment by snikch** [comment]
`@apollo/server` is included in the `package.json` so I don't think this is the issue. I created a fork and simply removed the `createApolloQueryValidationPlugin` function and it is working successfully, so I believe it's definitely the issue that `require('apollo-server-errors')` is not listed as a dependency.

**Comment by velias** [comment]
I do know nothing about next.js framework, so it is hard to determine. Is it and your project TypeScript or pure JavaScript? 
In this module's `package.json`, `apollo-server-core` is in `devDependency`, and `apollo-server-errors` is its transitive dependency. It is intentionally not in `dependencies` not to be included in projects not using Apollo3. But `devDependencies` from modules are typically not included. 
In the `index.js`, this `require('apollo-server-errors')` is intentionally in the code and not at the beginning, so should be used only if this code is used, but I'm not sure, maybe not. In `index.d.ts` it is imported at the beginning, so always. 
So I'd expect your project uses TS, and that import in types file is the problem.

Anyway, as I wrote before, for me the best solution should be to extract Apollo 3 plugin (and maybe even Envelop plugin) to separate files. But this is a breaking change, so I'd like to know @confuser opinion first, before I start to implement it.

**Comment by snikch** [comment]
Agree that the solution would be to have separate plugins!

**Comment by confuser** [comment]
@velias I think the best solve which is a larger change is to create a monorepo with a core folder containing the main business logic and then each plugin being its own package. Otherwise it'll mean installing a bunch of dependencies that aren't needed just to cover multiple use cases

**Comment by velias** [comment]
Agree, this is indeed the right solution. But it is a bit larger change ;-)

**Comment by lkrzyzanek** [comment]
Given that Apollo 3 is not supported anymore it could be possible to simply remove Apollo 3 plugin at all with all its dependencies.

**Comment by damienwebdev** [comment]
I've opened a [PR here](https://github.com/confuser/graphql-constraint-directive/pull/243) to address this. The fix is quite small, simply moving an import around to prevent unnecessary import (and attempted typecheck) on `apollo-server-errors` when using the `apollo4` import.

**Comment by SimplyKnownAsG** [comment]
The workaround mentioned by @davidruisinger is good, though be great to get that PR from @damienwebdev in! 100% thanks to the community though :wink:

**Comment by damienwebdev** [comment]
@confuser any chance we can get a release cut with this commit in it?

**Comment by jlczapski** [comment]
> @confuser any chance we can get a release cut with this commit in it?

+1 for this. Could really do with a more reliable fix..

**Comment by damienwebdev** [comment]
This is released as v6.0.0 for those who haven't noticed. Thanks @confuser
